packages/eth/src/websocket.rs

```diff
@@ -22,23 +28,78 @@ pub struct WebsocketClient {
     inner: HealthTrackingMiddleware<WsConnection>,
 }
 
+#[derive(Clone)]
+pub struct AwsClient {
+    client: KmsClient,
+}
+
+impl AwsClient {
+    pub fn try_new(
+        region: String,
+        access_key_id: String,
+        secret_access_key: String,
+        allow_http: bool,
+    ) -> ports::l1::Result<Self> {
```

Comment:
I'm not sure if this is only used for the e2e tests (in which case no problems), but if this is how we're initializing the AWS client when it's deployed, this won't work.

In a deployed environment, we should not be specifying manual credential providers. Both the standard AWS SDK, as well as rusoto, provide a default credential change (see https://github.com/rusoto/rusoto/blob/master/AWS-CREDENTIALS.md), and the expected behavior is to use that. In a deployed environment, EC2 instance profiles are used, which use a different flow than access keys.

Reply:
Actually, this might not work - we may need to move to Alloy. Seems like Rusoto doesn't support EKS-based authentication (see https://github.com/rusoto/rusoto/issues/1781).

Reply:
Ty for the heads-up. 

It seems like the auth interface is not sealed, so I may yet implement a auth chain that goes through all the methods the default auth provider uses and adds the EKS IRSA as the last.

We will also probably start on the Aloy move today, but I'd rather not have this blocked by that.

Reply:
Sounds good. If we're going to implement a custom auth provider chain, we should match (as much as reasonable) the default AWS credential order.

The Java SDK uses the following (which is the same in Python & Go, and the default AWS tries to push for): https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials-chain.html